controllers/binding/snapshotenvironmentbinding_controller.go

```diff
@@ -107,10 +109,13 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return helpers.HandleLoaderError(logger, err, "Snapshot", "SnapshotEnvironmentBinding")
 	}
 
-	component, err := loader.GetComponentFromSnapshot(r.Client, ctx, snapshot)
+	var component *applicationapiv1alpha1.Component
+	err = retry.OnError(retry.DefaultRetry, func(_ error) bool { return true }, func() error {
+		component, err = loader.GetComponentFromSnapshot(r.Client, ctx, snapshot)
```

Comment:
I can find a lot of error message `appstudio.openshift.io/component label missing from the Snapshot` in [prod log](https://rhcorporate.splunkcloud.com/en-US/app/search/search?q=search%20index%3D%22rh_rhtap_production_app%22%20%22kubernetes.namespace_name%22%3D%22integration-service%22%20%22structured.level%22%3D%22error%22%20%7C%20table%20structured.ts%20structured.caller%20structured.msg%20%20structured.error%20%7C%20sort%20%40timestamp&display.page.search.mode=verbose&dispatch.sample_ratio=1&workload_pool=standard_perf&earliest=-1d%40d&latest=now&display.page.search.tab=statistics&display.general.type=statistics&sid=1710393027.61616_7AF866D2-E4C8-447C-A52A-318F2E6F81AF) and look into `GetComponentFromSnapshot` and it returns error `fmt.Errorf("%s label missing from the Snapshot", gitops.SnapshotComponentLabel)` from https://github.com/redhat-appstudio/integration-service/blob/main/loader/loader.go#L142 when label is not found, which is unrecoverable and still causes the retry.
From my point of view, https://github.com/redhat-appstudio/integration-service/blob/main/loader/loader.go#L142 should return a new `NotFound` or `NotDefined` error, and update https://github.com/redhat-appstudio/integration-service/blob/main/helpers/errorhandlers.go#L69 to include this new error.
How do you think of it?

Reply:
Yes, I agree. We should instead throw a NewNotFound() error so that the `HandleLoaderError()` function doesn't requeue the failure indefinitely. And that's what I did [right here](https://github.com/redhat-appstudio/integration-service/pull/613/files#diff-8e9af1e483632eac17610237caaa5f879ea7bfb3e1e1ea228e45a17c770b9432R145), and now it doesn't requeue the errors, and just emits one log message ([as shown here](https://github.com/redhat-appstudio/integration-service/pull/613#discussion_r1521406156)) and stops reconciliation.

Wdyt?